File: code/cs/PlcDeviceConnection.Status.cs
// DOC
PlcStatus status = connection.Status;
status.Changed += HandleConnectionStatusChanged;

// DOC
private static void HandleConnectionStatusChanged(object sender, EventArgs e)
{
    var status = (PlcStatus)sender;

    Console.WriteLine(status.TimeStamp);
    Console.WriteLine("- Code=[{0}]", status.Code);
    Console.WriteLine("- Text=[{0}]", status.Text);
    Console.WriteLine("- Exception=[{0}]", status.Exception?.Message ?? "<none>");
}

File: code/vb/PlcDeviceConnection.Status.vb
'DOC
Dim status As PlcStatus = connection.Status
status.Changed += AddressOf HandleConnectionStatusChanged

'DOC
Private Shared Sub HandleConnectionStatusChanged(ByVal sender As Object, ByVal e As EventArgs)
    Dim status = CType(sender, PlcStatus)

    Console.WriteLine(status.TimeStamp)
    Console.WriteLine("- Code=[{0}]", status.Code)
    Console.WriteLine("- Text=[{0}]", status.Text)
    Console.WriteLine("- Exception=[{0}]", If(status.Exception?.Message, "<none>"))
End Sub
